lib/spack/spack/solver/asp.py

```diff
@@ -2189,7 +2189,7 @@ def build_specs(self, function_tuples):
         # concretization process)
         for root in self._specs.values():
             for spec in root.traverse():
-                if spec.version.is_commit:
+                if isinstance(spec.version, spack.version.GitVersion):
                     spec.version.generate_commit_lookup(spec.fullname)
```

Comment:
The name `generate_commit_lookup` seems to be a little too specific for the git variations now supported by this PR.  It might be better to rename it or add another layer of abstraction to account for ref types such as `git.branch` and `git.tags`